src/domain/flow-system/domain/src/entities/shared/flow_key.rs

```diff
@@ -43,13 +47,19 @@ impl FlowKey {
 pub struct FlowKeyDataset {
     pub dataset_id: DatasetID,
     pub flow_type: DatasetFlowType,
+    pub account_id: Option<AccountName>,
```

Comment:
I don't think that `account_id` is a property of flow key.
What will we do when sharing private datasets between several accounts? Each of them will need to see the flow. And we should see past flow runs on the datasets, which happened to datasets even before access to them was given. And similarly, we should not see future flow runs, or any runs, if the dataset access is lost.

So, we should allow listing flows for all accounts that **currently** have permissions to schedule it. In this case we do not need to record anything into flows that concerns accounts. Instead, we would use account id to query a list of "visible datasets", and it can be used to find a set flow keys that match it. The flow infrastructure would only have to support batching loads, ...

Reply:
Without it, we will not be able to index properly and we will not be able to make proper pagination.

Reply:
Removed